pkg/controller/discovery/controller.go

```diff
@@ -36,19 +36,23 @@ import (
 	"time"
 )
 
-var log = logging.WithName("discovery")
+var log logging.Logger
 
 // Application settings.
 var Settings = &settings.Settings
 
+func init() {
+	log = logging.WithName("discovery")
+	model.Log = &log
+	container.Log = &log
+	web.Log = &log
+}
```

Comment:
More appropriate to use package initialization to ensure the `Log` variables are set before they're used.